### DISCUSSION TITLE
10.1 RC

### DISCUSSION THREAD

**Discussion by mfazekas** [rnmapbox/maps#3228] [Announcements]
Released first RC version of 10.1

10.1 is a transitional release supporting both Mapbox 10.16* and Mapbox 11.*

Further releases are going to drop Mapbox 10.* support.

## Highlights of 10.1:
- Mapbox 11.0 support, including [Slots](https://rnmapbox.github.io/docs/components/SymbolLayer#slot), [StyleImport](https://rnmapbox.github.io/docs/components/StyleImport) configs, [Models](https://rnmapbox.github.io/docs/components/Models) and [ModelLayer](https://rnmapbox.github.io/docs/components/ModelLayer) (many thanks to the sponsor of this feature)
- new architecture/fabric support (many thanks to @WoLewicki and @j-piasecki huge contributions on this)
- dropped Maplibre and MapboxGL support (and cleaned up a lot of obsolete code)
- new [Viewport](https://rnmapbox.github.io/docs/components/Viewport) component, this replaces the deprecated [Camera#onUserTrackingModeChange](https://rnmapbox.github.io/docs/components/Camera#onusertrackingmodechange)
- new [LocationPuck](https://rnmapbox.github.io/docs/components/LocationPuck) component this replace the deprecated [UserLocation#renderMode](https://rnmapbox.github.io/docs/components/UserLocation)
- new [documentation site](https://rnmapbox.github.io)

## Removal/Deprecations:
- dropped Maplibre and MapboxGL support, RNMapboxMapsImpl is no longer needed
- UserLocation#renderMode `native` is deprecated use `LocationPuck` instead. In general we recommend using native `LocationPuck` instead of  javascript based`UserLocation`. The later still should work but `LocationPuck` should have better performance/smoothness.
- [Camera#onUserTrackingModeChange](https://rnmapbox.github.io/docs/components/Camera#onusertrackingmodechange) is deprecated and use [Viewport](https://rnmapbox.github.io/docs/components/Viewport). Also if you have complex transitions between following the user location, than consider adopting `Viewport`

## Acknowledgments and Sponsorship

A sincere thank you to our dedicated [sponsors](https://github.com/sponsors/rnmapbox), including our valued monthly sponsors, for their unwavering support of this repository. 🙏 The implementation of the Mapbox 11.* feature was made possible through individual sponsorship, where someone financially contributed to the development of this specific feature. We extend our deepest appreciation for this targeted support. 🙏

If you find value in this project, consider [sponsoring](https://github.com/sponsors/rnmapbox) it. While I genuinely appreciate the current level of sponsorship, sustaining the amount of time required for the project demands additional support. Your continued sponsorship ensures not only ongoing maintenance but also the capability to dedicate substantial time to the project's growth and improvement. Your increased support would be invaluable in propelling this open-source endeavor forward. Thank you for considering and being a diverse and crucial part of our community! 🚀

**Comment by RyanTG** [comment]
It looks like 10.1.0 was released today. But when  I do `expo install @rnmapbox/maps@next` it installs rc6. Am I doing something wrong?

Thanks for all your work (and the other contributors)! It’s very impressive.

**Reply by mfazekas** [reply]
The `@next` was only needed as 10.1 -rc-s was published with `next` tag. For released version you just want `expo install @rnmapbox/maps`
